stdlib/public/Concurrency/Task.swift

```diff
@@ -623,6 +632,81 @@ extension Task where Success == Never, Failure == Never {
       _enqueueJobGlobalWithDelay(duration, job)
     }
   }
+
+  /// Suspends the current task for _at least_ the given duration
+  /// in nanoseconds, unless the task is cancelled. If the task is cancelled,
+  /// throws \c CancellationError.
+  ///
+  /// This function does _not_ block the underlying thread.
+  public static func sleep(nanoseconds duration: UInt64) async throws {
+    // Allocate a word to store a flag indicating that someone has picked
+    // up the continuation, and initialize it to zero.
+    let wordPtr = UnsafeMutablePointer<Builtin.Word>.allocate(capacity: 1)
+    Builtin.atomicstore_seqcst_Word(
+        wordPtr._rawValue, UInt(0)._builtinWordValue)
+
+    // When we leave, remove the cancellation record that's used to detect
+    // when the task has been cancelled.
+    var cancellationRecord: UnsafeRawPointer? = nil
+    defer {
+      _taskRemoveCancellationHandler(record: cancellationRecord!)
```

Comment:
Does this remove the cancellation handler too soon? I can see the same thing being done in `withTaskCancellationHandler`, but there it’s deferred until after the operation is awaited. I think the cancellation handler should only be removed in `onWake` or `onCancel`.

Reply:
Ah - missed the await on `withUnsafeThrowingContinuation` 🤦 , so it should happen after the sleep has ended.

Reply:
I switched over to `withTaskCancellationHandler` to be certain that we got the ordering right.

Reply:
Thanks - wasn't sure if I'd missed something when I first looked.